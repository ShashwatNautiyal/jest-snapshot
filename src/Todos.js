import React from 'react';

const Todos = ({ todos }) => {
  if (!todos) {
    return <></>;
  }

  if (todos.length === 0) {
    return <span>Todo list is empty</span>;
  }

  return (
    <div>
      {todos.map((todo) => (
        <h2 key={todo}>{todo}</h2>
      ))}
    </div>
  );
};

export default Todos;
