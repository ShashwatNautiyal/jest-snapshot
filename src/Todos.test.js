import React from 'react';

import renderer from 'react-test-renderer';

import Todos from './Todos';

it("renders correctly when there are no todos", () => {
  const tree = renderer.create(<Todos />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there are empty todo", () => {
  const todos = [];
  const tree = renderer.create(<Todos todos={todos} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there are 3 todos", () => {
  const todos = ["Shopping", "Yoga class", "Doctor Appointment"];
  const tree = renderer.create(<Todos todos={todos} />).toJSON();
  expect(tree).toMatchSnapshot();
});
