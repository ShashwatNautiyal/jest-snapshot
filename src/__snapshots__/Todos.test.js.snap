// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are 3 todos 1`] = `
<div>
  <h2>
    Shopping
  </h2>
  <h2>
    Yoga class
  </h2>
  <h2>
    Doctor Appointment
  </h2>
</div>
`;

exports[`renders correctly when there are empty todo 1`] = `
<span>
  Todo list is empty
</span>
`;

exports[`renders correctly when there are no todos 1`] = `null`;
